include/lbann/base.hpp

```diff
@@ -163,10 +163,16 @@ static bool __attribute__((used)) endsWith(const std::string mainStr, const std:
 }  // namespace lbann
 
 /// Print the dimensions and name of a Elemental matrix
-static void __attribute__((used)) _display_matrix(ElMat *m, const char *name) {
+static void __attribute__((used)) _print_matrix_dims(AbsDistMat *m, const char *name) {
```

Comment:
IIRC, `__attribute__` is a GCC extension and not ISO C++. Also, why internal linkage in a file that will be `#include`d in every translation unit in the project and in every translation unit of downstreams?